source/Module/STIG/Functions.Checklist.ps1

```diff
@@ -4,128 +4,217 @@
+        [ValidateScript(
+        {
+            If (Test-Path -Path $_ -PathType Leaf)
+            {
+                Return $True
+            } else {
+                Throw "$($_) is not a valid path to a reference configuration (.mof) file. Provide a full valid path and filename."
+            }
+        }
+        )]
+        [String]
         $ReferenceConfiguration,
 
-        [Parameter(Mandatory = $true, ParameterSetName = 'result')]
-        [psobject]
-        $DscResult,
+        [Parameter(Mandatory = $true, ParameterSetName = 'dsc')]
+        [PSObject]
+        $DscResults,
 
         [Parameter(Mandatory = $true)]
-        [string]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript(
+        {
+            foreach ($filename in $_)
+            {
+                If (Test-Path -Path $filename -PathType Leaf)
+                {
+                    Return $True
+                } else {
+                    Throw "$($filename) is not a valid path to a DISA STIG .xccdf file. Provide a full valid path and filename."
+                }
+            }
+        }
+        )]
+        [String[]]
         $XccdfPath,
 
-        [Parameter(Mandatory = $true)]
-        [System.IO.FileInfo]
-        $OutputPath,
-
         [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript(
+        {
+            If (Test-Path -Path $_ -PathType Leaf)
+            {
+                Return $True
+            } else {
+                Throw "$($_) is not a valid path to a ManualChecklistEntries.xml file. Provide a full valid path and filename."
+            }
+        }
+        )]
         [String]
-        $ManualCheckFile
-    )
+        $ManualChecklistEntries,
 
-    # Validate parameters before continuing
-    if ($ManualCheckFile)
-    {
-        if (-not (Test-Path -Path $ManualCheckFile))
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript(
         {
-            throw "$($ManualCheckFile) is not a valid path to a ManualCheckFile. Provide a full valid path"
+            If (Test-Path -Path $_.DirectoryName -PathType Container)
+            {
+                Return $True
+            } else {
+                Throw "$($_) is not a valid directory. Please provide a valid directory."
+            }
+            if ($_.Extension -ne '.ckl')
+            {
+                throw "$($_.FullName) is not a valid checklist extension. Please provide a full valid path ending in .ckl"
+            } else {
+                Return $True
+            }        
         }
-        [string]$manualCheckData = Get-Content $manualCheckFile
-    }
-
-    if (-not (Test-Path -Path $OutputPath.DirectoryName))
-    {
-        throw "$($OutputPath.DirectoryName) is not a valid directory. Please provide a valid directory."
-    }
+        )]
+        [System.IO.FileInfo]
+        $OutputPath
+    )
 
-    if ($OutputPath.Extension -ne '.ckl')
+    if ($ManualChecklistEntries)
```

Comment:
if this is detecting if $ManualChecklistEntries was used / passed to the function, then use the ContainsKey method of PSBoundParameters automatic variable here.
i.e.:
$PSBoundParameters.ContainsKey('ManualChecklistEntries')

Reply:
Done.